spirv_common.hpp

```diff
@@ -343,14 +343,16 @@ struct SPIREntryPoint
 {
 	SPIREntryPoint(uint32_t self_, spv::ExecutionModel execution_model, std::string entry_name)
 	    : self(self_)
-	    , name(std::move(entry_name))
+	    , name(entry_name)
+	    , orig_name(entry_name)
 	    , model(execution_model)
 	{
 	}
 	SPIREntryPoint() = default;
 
 	uint32_t self = 0;
 	std::string name;
+	std::string orig_name;
```

Comment:
So if I understand it correctly, orig_name will be used for "set_entry_point()" and friends, but name is remapped as necessary?

Reply:
No. The entry point renaming is universal.

The only thing `orig_name` is used for is extracting the mapping via the `get_entry_point_name_map()` API function, per issue #316.

Reply:
So orig_name is recorded during parse, but name is the remapped name which happens during compile() in case the entry point name is incompatible?

Reply:
Yes...illegal function names...both in the entry point and the function alias are renamed early in the compile stage.

The `orig_name` is retained strictly to allow a post-compile query to be made about which names were changed and from what to what.